packages/components/src/experience-control/index.tsx

```diff
@@ -1,98 +1,132 @@
-import { Button } from '@wordpress/components';
+import { BaseControl, useBaseControlProps, VisuallyHidden } from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
 import clsx from 'clsx';
+import { useId } from 'react';
 import { IconBad, IconGood, IconNeutral } from './icons';
+import type {
+	ExperienceValue,
+	ExperienceOption,
+	ExperienceControlOptionProps,
+	ExperienceControlBaseProps,
+	ExperienceControlProps,
+} from './types';
 
 import './style.scss';
 
-enum Experience {
-	GOOD = 'good',
-	NEUTRAL = 'neutral',
-	BAD = 'bad',
-}
-
 const ExperienceControlOption = ( {
 	className,
-	isSelected,
-	onClick,
-	children,
-}: {
-	className?: string;
-	isSelected: boolean;
-	onClick: () => void;
-	children: React.ReactNode;
-} ) => (
-	<Button
-		className={ clsx( 'experience-control__button', className, {
-			'is-selected': isSelected,
-		} ) }
-		onClick={ onClick }
-	>
-		<div className="experience-control__button-content">{ children }</div>
-	</Button>
+	label,
+	ariaLabel,
+	...restProps
+}: ExperienceControlOptionProps ) => (
+	<label className={ clsx( 'a8c-experience-control__option', className ) }>
+		<input
+			type="radio"
+			className="a8c-experience-control__radio"
+			aria-label={ ariaLabel }
+			{ ...restProps }
+		/>
+		<div className="a8c-experience-control__option-label">{ label }</div>
+	</label>
```

Comment:
The use of `label` and `aria-label` here are a bit unclear to me. `label` implies a string, and it's non-obvious whether `label` or `aria-label` is going to take precedence as the actual label of the input. Everything being wrapped in a `<label>` element also strongly implies that `label` is expected to be a text node, or at least have a text representation.

I can think of a few ways around this weirdness, for example we can expect the consumer to pass in an already accessibly-labeled `label` node, instead of a separate aria label that is passed on to the `input`.

Reply:
Agree.

Do you think we can use `icon` instead of `label` something like:

```
const ExperienceControlOption = ( {
	className,
	icon,
	ariaLabel,
	...restProps
}: ExperienceControlOptionProps ) => (
	<label className={ clsx( 'a8c-experience-control__option', className ) }>
		<VisuallyHidden>
			<input type="radio" aria-label={ ariaLabel } { ...restProps } />
		</VisuallyHidden>
		<div className="a8c-experience-control__option-icon" aria-hidden="true">
			{ icon }
		</div>
	</label>
);
```

Reply:
Sure, that's probably fine for now, since we only have this one use case. In the future when this abstraction needs to support more use cases, it may be too limiting since it won't necessarily be an icon (e.g. an NPS score selector where it's just plain text numbers). But we don't have to overthink it at this moment.